src/GitHub.App/Services/TeamExplorerContext.cs

```diff
@@ -40,17 +43,18 @@ public class TeamExplorerContext : ITeamExplorerContext
         ILocalRepositoryModel repositoryModel;
 
         [ImportingConstructor]
-        public TeamExplorerContext(IGitHubServiceProvider serviceProvider, IVSGitExt gitExt)
-            : this(gitExt, serviceProvider)
+        TeamExplorerContext(SVsServiceProvider serviceProvider, IVSGitExt gitExt)
```

Comment:
If this were imported using `[Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider` then I think these two constructors could be merged?

Reply:
We do need to allow a `JoinableTaskFactory` to be specified when unit testing. If we exported it with a `ContractName`, we could combine that with what you suggested, e.g:

```
[Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider,
[Import("GitHub.VisualStudio")] JoinableTaskFactory joinableTaskFactory
```
(or perhaps `JoinableTaskContext`)